src/System.Net.Sockets/src/System/Net/Sockets/Socket.cs

```diff
@@ -114,6 +114,11 @@ public Socket(AddressFamily addressFamily, SocketType socketType, ProtocolType p
             _socketType = socketType;
             _protocolType = protocolType;
 
+            if (addressFamily == AddressFamily.InterNetworkV6)
+            {
+                DualMode = false;
```

Comment:
> This shouldn't be necessary. If for some reason it is for *NIX, we should move it within the PAL. 

The initial value of IPV6_V6ONLY may differ between platforms. I thought about moving it behind the PAL and ended up leaving it here for clarity. I'll change this to a `Debug.Assert()` and move the setup behind the PAL.